crates/bevy_hierarchy/src/one_to_one/component.rs

```diff
@@ -0,0 +1,189 @@
+#[cfg(feature = "reflect")]
+use bevy_ecs::reflect::{
+    ReflectComponent, ReflectFromWorld, ReflectMapEntities, ReflectVisitEntities,
+    ReflectVisitEntitiesMut,
+};
+use bevy_ecs::{
+    component::{Component, ComponentId},
+    entity::{Entity, VisitEntities, VisitEntitiesMut},
+    event::Events,
+    traversal::Traversal,
+    world::{DeferredWorld, FromWorld, World},
+};
+use core::{fmt::Debug, marker::PhantomData, ops::Deref};
+
+use super::OneToOneEvent;
+
+/// Represents one half of a one-to-one relationship between two [entities](Entity).
```

Comment:
It might be reasonable to mention that this is a symmetric or undirected one-to-one relationship. After adding the component, you can't tell which points to which.

There is also a need for `OneToOne` relationships that are directed, or not symmetric. Like:
```
struct Train;
type PreviousCar = OneToOneSource<Train>;
type NextCar = OneToOneTarget<Train>;
```
A single entity may be at the same time the source and the target of a relation towards different objects.

Reply:
I have added directed `OneToOne` relationships, along with `ManyToMany` as well.